### ISSUE TITLE
'packer build -var x=y' regression from 1.1.3?

### ISSUE THREAD

**Issue by tjohnston01** [hashicorp/packer#5926]
- I think I am seeing a regression from 1.1.3 (when I build my regular templates using the released **1.2.0** binary). 
- My build works if I swap the `packer` binary back to the 1.1.3 release build.
- I am running `packer` inside a Docker container based on the `ubuntu:xenial` image.

See [here](https://gist.github.com/tjohnston01/896eb97f18d3c3bf6f5657d820749690) for the debug log. I sanitized some paths, but otherwise this log is untouched.

[Here](https://gist.github.com/tjohnston01/0ed62e75c8328111713487c939dc8745) is a simple template to reproduce the issue. It should be used with this command (vars providing the ESXi build datastore/host/password/username):
`packer build -var datastore=... -var remote=... -var remoteuser=... -var password=... test.json`

**Question:** Can `-var` still be used the same way as before with the `build` command?
If I remove all the `-var` arguments from my build command, the build does not exhibit this crash (fails at the template parsing stage because a file path is not templated by the user var it expects).

In my template, I have variables expanded like this:
```
"iso_url": "{{user `workspace`}}/IMAGE_DEPLOY_SCRIPTS/ISO/build/vm-xenial-minimal-amd64-xenial-netboot/vm-xenial-minimal-amd64.iso",
```

Maybe a similar issue to https://github.com/hashicorp/packer/issues/5921 ?

**Comment by SwampDragons** [comment]
Looking at your logs, I think this one's actually caused by #5830 and should be fixed by https://github.com/hashicorp/packer/commit/a4123eabe35e3d29e3aa6d50efa107800aa87cb6. So this'll be fixed in 1.2.1.  Sorry for the regression; you can either revert to 1.1.3 for now or build 1.2.1-dev from source.

**Comment by tjohnston01** [comment]
Trying this now with the 1.2.1 released binary, and it seems to be fixed.
Thanks!
